subprojects/docs/src/docs/userguide/dependencyManagementJavaProjects.adoc

```diff
@@ -0,0 +1,109 @@
+    <sourcefile file="build.gradle" snippet="define-dependency"/>
+</sample>
+++++
+
+To find out more about defining dependencies, have a look at <<declaring_dependencies>>.
+
+[[sec:configurations_java_tutorial]]
+=== Using dependency configurations
+
+A api:org.gradle.api.artifacts.Configuration[] is a named set of dependencies and artifacts. There are three main purposes for a _configuration_:
+
+Declaring dependencies::
+The plugin uses configurations to make it easy for build authors to declare what other subprojects or external artifacts are needed for various purposes during the execution of tasks defined by the plugin.
+Resolving dependencies::
+The plugin uses configurations to find (and possibly download) inputs to the tasks it defines.
+Exposing artifacts for consumption::
+The plugin uses configurations to define what _artifacts_ it generates for other projects to consume.
+
+With those three purposes in mind, let's take a look at a few of the <<sec:java_library_configurations_graph,standard configurations defined by the Java Library Plugin>>.
+
+implementation::
+The dependencies required to compile the production source of the project, but which are not part of the API exposed by the project. This configuration is an example of a configuration used for declaring dependencies.
+runtimeClasspath::
+The dependencies required by the production classes at runtime. By default, this includes the dependencies declared in the `api`, `implementation`, and `runtimeOnly` configurations. This configuration is an example of a configuration used for resolving dependencies, and as such, users should never declare dependencies directly in the `runtimeClasspath` configuration.
+apiElements::
+The dependencies which are part of this project's externally consumable API as well as the classes which are defined in this project which should be consumable by other projects. This configuration is an example of exposing artifacts for consumption.
+
+Various plugins add further standard configurations. You can also define your own custom configurations in your build via api:org.gradle.api.Project#configurations[]. See <<defining_and_using_configurations>> for the details of defining and customizing dependency configurations.
+
+[[sec:repositories_java_tutorial]]
+=== Declaring repositories
+
+How does Gradle know where to find the files for external dependencies? Gradle looks for them in a _repository_. A repository is a collection of modules, organized by `group`, `name` and `version`. Gradle understands different <<repository_types,repository types>>, such as Maven and Ivy, and supports various ways of accessing the repository via HTTP or other protocols.
+
+By default, Gradle does not define any repositories. You need to define at least one with the help of api:org.gradle.api.Project#repositories[] before you can use module dependencies. One option is use the Maven Central repository:
```

Comment:
I am still of the opinion that most people rather know Maven Central than JCenter. This is especially useful for user migrating from Maven. They usually do not use JCenter so they can feel right at home. I have nothing against sprinkling JCenter into the content here and there but in the end it should be the user's call what they want to use here. IHMO it doesn't make a big difference in terms of functionality. 

I have seen JCenter having issues as well in the past when downloading artifacts. If we think JCenter is better then I'd like to link to an official comparison (with data) so we can back up the claim that JCenter is generally better.

> Using Maven Central encourages it's use, and jcenter() is strictly better here.

I don't think I understand this comment. Can you clarify?

Reply:
>> Using Maven Central encourages it's use, and jcenter() is strictly better here.

> I don't think I understand this comment. Can you clarify?

Folks coming from Maven should use `jcenter()` because it's a complete mirror, and my understanding is it's much more stable and faster than Maven Central. It should be the users call, of course, but we want to encourage the better thing by using it in our examples IMO.